sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```diff
@@ -81,6 +81,7 @@ class Iso8601DateFormatter(
     try {
       formatter
     } catch checkLegacyFormatter(pattern, legacyFormatter.validatePatternString)
+    ()
```

Comment:
ERROR] [Error] /spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala:83: type mismatch;
 found   : PartialFunction[Throwable,java.time.format.DateTimeFormatter]
 required: Throwable => Unit
[INFO] [Info] : PartialFunction[Throwable,java.time.format.DateTimeFormatter] <: Throwable => Unit?
[INFO] [Info] : false

Reply:
This aligns with Scala 3. I guess the `Unit` here comes from the method result type. I'll ask whether that is desirable; arguably it could insert the parens automatically.

Reply:
thx ~ @som-snytt